### PR TITLE
getLogger api documentation changes

### PR DESCRIPTION
**Pull request by anirudh2290**
log.getLogger api documentation changes

@mli, @zackchase, @madjam, @nswamy

### CODE REVIEW

File: python/mxnet/log.py [hunk 1]
```diff
@@ -63,12 +63,38 @@ def format(self, record):
 def getLogger(name=None, filename=None, filemode=None, level=WARNING):
     """Get customized logger.
 
-    Args:
-        name: Name of the logger.
-        level: Level to log.
+    Parameters
+    ----------
+    name: str, optional
+        Name of the logger
```
Comment (pracheer) [inline python/mxnet/log.py]: Minor: The body paragraph for all the parameters and return objects should end in a '.' (period)

File: python/mxnet/log.py [hunk 2]
```diff
@@ -63,12 +63,38 @@ def format(self, record):
 def getLogger(name=None, filename=None, filemode=None, level=WARNING):
     """Get customized logger.
```
Comment (pracheer) [inline python/mxnet/log.py]: '"Gets a" customer logger' or may be '"Returns a" customer logger' (preferred I think).

File: python/mxnet/log.py [hunk 3]
```diff
@@ -63,12 +63,38 @@ def format(self, record):
...
 
-    Args:
-        name: Name of the logger.
-        level: Level to log.
+    Parameters
+    ----------
+    name: str, optional
+        Name of the logger
+    filename: str, optional
+        The filename to which logging output will be sent
```
Comment (pracheer) [inline python/mxnet/log.py]: nitpick: ..to which the "logger's output" will be "written".

File: python/mxnet/log.py [hunk 4]
```diff
@@ -63,12 +63,38 @@ def format(self, record):
...
+        default is 'a' if `filename` is not ``None``
+    level: int, optional
+        The `logging` level for the logger
+        See: https://docs.python.org/2/library/logging.html#logging-levels
 
-    Returns:
-        A logger.
+    Returns
+    -------
+    `Logger` object
```
Comment (pracheer) [inline python/mxnet/log.py]: This line could just be written as:
Logger
(Backticks are needed only for types that are referenced in paragraphs.)

**Comment by anirudh2290** [comment]
@pracheer  Thank you for reviewing. I have made the changes that you requested.

**Comment by nswamy** [comment]
@piiswrong Please merge this, comments are addressed

File: python/mxnet/log.py [hunk 5]
```diff
@@ -61,14 +61,40 @@ def format(self, record):
...
+    -------
+    >>> from mxnet.log import getLogger
+    >>> logger = getLogger("Test")
+    >>> logger.warn("Hello World")
+    W0505 00:29:47 3525 <stdin>:<module>:1] Hello World
+    >>> import logging
+    >>> logger = getLogger("Test2", level=logging.WARNING)
+    >>> logger.warn("Hello World")
+    W0505 00:30:50 3525 <stdin>:<module>:1] Hello World
+    >>> logger.debug("Hello World")
```
Comment (piiswrong) [inline python/mxnet/log.py]: These are 3 different examples.
separate them with comments explaining what each example is trying to show

**Comment by ZihengJiang** [comment]
How about changing it to `get_logger` for python coding style?

**Comment by piiswrong** [comment]
get_logger is better. Though I'm not sure how many people are using this. probably leave an alias and mark as deprecated.

**Comment by anirudh2290** [comment]
@piiswrong I have made the necessary changes.

**Pull request merged**